### PR TITLE
Don't require systemd for `modify-config`

### PR DESCRIPTION
**Pull request by superm1**
fwupdmgr has the ability to restart the daemon itself by using `fu_util_quit` to stop it and dbus activation to start it again.

Use that instead.

Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [x] Feature
- [ ] Documentation

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -3672,10 +3672,12 @@ fu_util_modify_config(FuUtilPrivate *priv, gchar **values, GError **error)
 					   _("Restart the daemon to make the change effective?")))
 			return TRUE;
 	}
-#ifdef HAVE_SYSTEMD
-	if (!fu_systemd_unit_stop(fu_util_get_systemd_unit(), error))
+
+	if (!fu_util_quit(priv, NULL, error))
 		return FALSE;
-#endif
+	if (!fwupd_client_connect(priv->client, priv->cancellable, error))
```
Comment (hughsie) [inline src/fu-util.c:3678]: does this work, or do you get to the:

	/* nothing to do */
	if (priv->proxy != NULL) {
		g_task_return_boolean(task, TRUE);
		return;
	}

case?

Comment (superm1) [reply src/fu-util.c:3678]: Ah... it's hitting that case, that explains why it returned so quickly.  An alternative is to change the message and just "Quit" the daemon.

Comment (superm1) [reply src/fu-util.c:3678]: Pushed a fix.

File: libfwupd/fwupd-client.c [hunk 2]
```diff
@@ -1058,6 +1058,7 @@ fwupd_client_quit_async(FwupdClient *self,
 			  cancellable,
 			  fwupd_client_quit_cb,
 			  g_steal_pointer(&task));
+	g_clear_object(&priv->proxy);
```
Comment (hughsie) [inline libfwupd/fwupd-client.c:1061]: i think this needs to be *in* fwupd_client_quit_cb(), i.e. just after the `/* success */`

Comment (superm1) [reply libfwupd/fwupd-client.c]: I put it in `fwupd_client_quit_finish` as that seemed more sensible to me.  If you think it needs to be in `fwupd_client_quit_cb` LMK.

Comment (hughsie) [reply libfwupd/fwupd-client.c]: > as that seemed more sensible to me

Maybe this isn't a concern for the CLI, but I'm worried about what *thread* and GMainContext is unreffing the proxy. I think fwupd_client_quit_cb has a lesser chance of exploding in our future faces.

Comment (superm1) [reply libfwupd/fwupd-client.c]: pushed a fix

Comment (hughsie) [reply libfwupd/fwupd-client.c]: I pushed a nit-fix -- apologies!

**Review by hughsie — DISMISSED** [review]

File: libfwupdplugin/fu-efi-section.c [hunk 3]
```diff
@@ -324,6 +324,17 @@ fu_efi_section_parse(FuFirmware *firmware,
...
+		   g_strcmp0(fu_firmware_get_id(firmware), FU_EFI_SECTION_GUID_TIANO_COMPRESS) ==
+		       0) {
+		g_error("MOO");
+		if (!fu_efi_section_parse_compression_sections(self,
+							       partial_stream,
+							       flags,
+							       error)) {
+			g_prefix_error(error, "failed to parse lzma section: ");
+			return FALSE;
+		}
```
Comment (superm1) [inline libfwupdplugin/fu-efi-section.c:337]: huh, what's this?

Comment (hughsie) [reply libfwupdplugin/fu-efi-section.c]: evidence that it's time to go to bed. fixed!

**Review by hughsie — APPROVED** [review]

File: libfwupd/fwupd-client.c [hunk 4]
```diff
@@ -1007,6 +1007,8 @@ fwupd_client_disconnect(FwupdClient *self, GError **error)
 static void
 fwupd_client_quit_cb(GObject *source, GAsyncResult *res, gpointer user_data)
 {
+	FwupdClient *self = FWUPD_CLIENT(g_task_get_source_object(G_TASK(user_data)));
```
Comment (superm1) [inline libfwupd/fwupd-client.c:1010]: This isn't a memory leak now?  I was worried that the g_autoptr(GTask) is using the same reference.

Comment (hughsie) [reply libfwupd/fwupd-client.c:1010]: nah, i think it's fine -- just two things pointing to the same address

**Pull request merged**